### DISCUSSION TITLE
Output audio to AM radio

### DISCUSSION THREAD

**Discussion by ghost** [pschatzmann/arduino-audio-tools#974] [Q&A] [answered]
Hello,

I'm interested to output audio to AM radio on esp32.  

This arduino library does that https://github.com/bitluni/ESP32AMRadioTransmitter

I wonder if this is possible?  If yes, if you point me in the right direction of how to connect these two libraries then I can have a go at writing the code.  Just a hint at the basic approach would be a huge help.

thanks!

**Comment by pschatzmann** [comment]
Did you test this? 
The last time I tried, it did not work with the actual ESP32 release any more...

**Reply by ghost** [reply]
Yes I just got it working a few minutes ago, though I'm puzzled because the audio seems to play a little slow - thus people's voices sound kind of deep/slurring not sure why.

The instructions on the github page say to use pin 25 for output but it is actually pin 26.

FYI I just copied the code into my project - it seems to be less than 50 lines.

If you felt inclined to build it into your Arduino Audio Library project that would be doubly awesome!  Official code would be better than what I hack together no doubt.

**Comment by pschatzmann** [comment, marked as answer]
You would redesign the [ESP32AMRadioTransmitter](https://github.com/bitluni/ESP32AMRadioTransmitter) to support a stream of data as input instead of an array.

**Reply by ghost** [reply]
Maybe I'll look at the source code of some of your sinks and see how they do it.

**Reply by ghost** [reply]
Can you suggest any particular parts of your code base that I should study that are likely to be most similar to what is needed? thanks

**Comment by pschatzmann** [comment]
As I saied I would start to redesign  the RadioTransmitter to provide a write (const uint8_t *buffer, size_t size) method. It should support the buffer to be int16_t (bits_per_sample == 16).  VolumeOutput (or any other class ending with Output) is a simple output only class. 

But I would redesign the current code in several steps: first I would change it to support the current int8_t data so that you can test your changes with the current data and then change it to int16_t in a second step.

**Reply by ghost** [reply]
Thanks for the starting hints.  I have found a different technical solution to my problem but may come back to this in the future.
